src/test/unit/sanitize.test.js

```diff
@@ -204,4 +206,78 @@ describe('sanitizePII', function() {
       expect(extensions.baseURL.length).toEqual(0);
     }
   });
+
+  it('should not sanitize all the preference values inside preference read markers', function() {
+    const profile = getProfileWithMarkers([
+      [
+        'PreferenceRead',
+        1,
+        {
+          type: 'PreferenceRead',
+          startTime: 0,
+          endTime: 1,
+          prefAccessTime: 0,
+          prefName: 'preferenceName',
+          prefKind: 'preferenceKind',
+          prefType: 'preferenceType',
+          prefValue: 'preferenceValue',
+        },
+      ],
+    ]);
+    const PIIToRemove = getRemoveProfileInformation({
+      shouldRemovePreferenceValues: false,
+    });
+    const sanitizedProfile = sanitizePII(profile, PIIToRemove).profile;
+
+    expect(sanitizedProfile.threads.length).toEqual(1);
+
+    const thread = sanitizedProfile.threads[0];
+    expect(thread.markers.length).toEqual(1);
+
+    const marker = thread.markers.data[0];
+    // All the conditions have to be checked to make Flow happy.
```

Comment:
In tests we sometimes use shortcuts and use `any`, like `expect((marker: any).prefValue).toEqual('preferenceValue')`. It's less a big deal in tests to use `any` especially for expectations.

But I think this is good as it is too so I'll just merge this, thanks :)

Reply:
Sadly, this is exactly the approach I was trying to *avoid*. I knew that would work but since it's a total subversion of the type checker, I was trying not to use it. The approach I took *feels* better to me but the next time I will go with the "easier" version. Thanks again for all the mentorship on landing this!